packages/eslint-plugin-turbo/lib/utils/calculate-inputs.ts

```diff
@@ -0,0 +1,449 @@
+
+  // After processing length is 0, 1, or more than 1.
+  switch (processed.length) {
+    case 0:
+      return [];
+    case 1:
+      return processed;
+    default:
+      return [...new Set(processed)].sort();
+  }
+}
+
+function processEnv(env: string[] | undefined): string[] {
+  if (!env) {
+    return [];
+  }
+
+  switch (env.length) {
+    case 0:
+      return [];
+    case 1:
+      return [env[0]];
+    default:
+      return [...new Set(env)].sort();
+  }
+}
+
+function processPassThroughEnv(
+  passThroughEnv: string[] | null | undefined
+): string[] | null {
+  if (!passThroughEnv) {
+    return null;
+  }
+
+  switch (passThroughEnv.length) {
+    case 0:
+      return [];
+    case 1:
+      return [passThroughEnv[0]];
+    default:
+      return [...new Set(passThroughEnv)].sort();
+  }
+}
+
+function processDotEnv(
+  workspacePath: string,
+  filePaths: string[] | null | undefined
+): DotEnvConfig | null {
+  if (!filePaths) {
+    return null;
+  }
+
+  const hashEntries: [string, string][] = [];
+  filePaths.reduce((accumulator, filePath) => {
+    const hash = crypto.createHash("sha1");
+    try {
+      const fileContents = fs.readFileSync(path.join(workspacePath, filePath));
+      hash.update(fileContents);
+      accumulator.push([filePath, hash.digest("hex")]);
+    } catch (_) {}
+
+    return accumulator;
+  }, hashEntries);
+
+  return {
+    filePaths,
+    hashes: Object.fromEntries(hashEntries),
+  };
+}
+
+// Generate `EnvironmentConfig`s
+
+function processGlobal(
+  workspacePath: string,
+  rootTurboJson: RootSchema
+): EnvironmentConfig {
+  return {
+    legacyConfig: processLegacyConfig(rootTurboJson.globalDependencies),
+    env: processEnv(rootTurboJson.globalEnv),
+    passThroughEnv: processPassThroughEnv(rootTurboJson.globalPassThroughEnv),
+    dotEnv: processDotEnv(workspacePath, rootTurboJson.globalDotEnv),
+  };
+}
+
+function processTask(workspacePath: string, task: Pipeline): EnvironmentConfig {
+  return {
+    legacyConfig: processLegacyConfig(task.dependsOn),
+    env: processEnv(task.env),
+    passThroughEnv: processPassThroughEnv(task.passThroughEnv),
+    dotEnv: processDotEnv(workspacePath, task.dotEnv),
+  };
+}
+
+const TEST_FALSE = (_: string): boolean => false;
+function generateEnvironmentTest(
+  config: EnvironmentConfig,
+  workspacePath: string | undefined
+): EnvironmentTest {
+  const output: EnvironmentTest = {
+    legacyConfig: TEST_FALSE,
+    env: TEST_FALSE,
+    passThroughEnv: TEST_FALSE,
+    dotEnv: TEST_FALSE,
+  };
+
+  if (config.legacyConfig.length > 0) {
+    const dependsOnEnvSet = new Set(config.legacyConfig);
+    output.legacyConfig = (variable: EnvVar) => dependsOnEnvSet.has(variable);
+  }
+
+  if (config.env.length > 0) {
+    const testables = wildcardTests(config.env);
+    output.env = (variable: EnvVar) => {
+      return (
+        testables.inclusions.test(variable) &&
+        !testables.exclusions.test(variable)
+      );
+    };
+  }
+
```

Comment:
Somewhere right around here there is a missing feature for framework detection.

Since this is now workspace-aware we can do it pretty-easily.